src/platform/react-hooks/src/hooks/useAbly.ts

```diff
@@ -2,8 +2,8 @@ import React from 'react';
 import { getContext } from '../AblyProvider.js';
 import * as API from '../../../../../ably.js';
 
-export function useAbly(id = 'default'): API.AbstractRealtime {
-  const client = React.useContext(getContext(id)) as API.AbstractRealtime;
+export function useAbly(id = 'default'): API.RealtimeInterface {
```

Comment:
Is there a reason to use `RealtimeInterface` over just using the `Realtime` class in the typing? I think having the word 'interface' in the name of an interface is often considered a code smell so it would be good to avoid having this in an important class in our typings, could we get away with only exposing the realtime class itself and letting people use that as the typing for any realtime client?

Reply:
The `Realtime` class is only exported by the default variant of the library. Presumably we want people to also be able to use React hooks with the modular variant of the library, which exports the `BaseRealtime` class instead. `RealtimeInterface` is the common interface that both of these classes conform to.

Reply:
ok yeah that makes sense, so they're basically the same type but not compatible because of the constructor typing mismatch? In that case it's fine to keep this but i would prefer `RealtimeClient` over `RealtimeInterface` and to avoid using 'interface' as a suffix for interface names in the public API if possible

Reply:
Sure, have changed to `*Client`.